### DISCUSSION TITLE
Hyphenation is wrong (at least in Italian texts)

### DISCUSSION THREAD

**Discussion by GiovanniSalmeri** [spacecowboy/Feeder#323] [General]
First of all, congratulations for this excellent program! Some years ago it was already very good, now it is pretty much perfect.

I would like to point out just a little glitch in Italian (and perhaps in all non-English) texts: the hyphenation applied is wrong, I suspect that the English patterns are applied even if the feed contains the tag `<language>it-IT</language>` which specifies the language. This is not a big deal, but perhaps also the solution is easy.

Thank you very much!

**Comment by spacecowboy** [comment]
good to know. but i suspect the hyphenation is quite hard coded. it has no language option that i know of at least

**Reply by GiovanniSalmeri** [reply]
I have given a quick glance at the source code... If this is the class used: https://developer.android.com/reference/kotlin/androidx/compose/ui/text/style/Hyphens, it should be locale-aware ("Automatic hyphenation is added between syllables at appropriate hyphenation points, following language rules"). Perhaps the solutions is somewhere here https://developer.android.com/reference/kotlin/androidx/compose/ui/text/intl/Locale ? Sorry if I'm saying something useless, unfortunately my knowledge of Android and Kotlin is zero :-(

**Reply by spacecowboy** [reply]
Right. But the current locale is not necessarily the same as the language text. So there's one reason why it might seem off at times
